### PR TITLE
Load routes in ErrorHandlerMiddleware.

### PR DESCRIPTION
**Pull request by ADmad**
This ensures app URLs on error pages are generated properly if an exception occurs before RoutingMiddleware is run.

### CODE REVIEW

**Comment by ADmad** [comment]
I'll add tests if the change seems good.

**Comment by markstory** [comment]
What about the solution done in 26bad8fda726b5813fb31f0dc6e63c0558c87691 ? Do we need this in addition to that change?

**Comment by ADmad** [comment]
The commit you linked just ensures that the router gets the proper request context (for proper base url). That still accounts only for static asset URLs.

Loading the routes still needs to be done so that app URLs are properly generated on error pages, else you get missing route errors. This PR takes care of that.

File: src/Error/Middleware/ErrorHandlerMiddleware.php [hunk 1]
```diff
@@ -244,4 +259,30 @@ protected function getExceptionTrap(): ExceptionTrap
...
+
+        try {
+            $builder = Router::createRouteBuilder('/');
+
+            $this->app->routes($builder);
+            if ($this->app instanceof PluginApplicationInterface) {
+                $this->app->pluginRoutes($builder);
+            }
+        } catch (Throwable $e) {
+            // Continue to avoid recursive exceptions.
```
Comment (markstory) [inline src/Error/Middleware/ErrorHandlerMiddleware.php]: Should we trigger a warning or log here?

Comment (ADmad) [reply src/Error/Middleware/ErrorHandlerMiddleware.php]: I guess we could trigger a warning, wouldn't want to fill up the logs with nested errors.

**Pull request merged**